cliquet/utils.py

```diff
@@ -123,5 +123,9 @@ def current_service(request):
     if request.matched_route:
         services = request.registry.cornice_services
         pattern = request.matched_route.pattern
-        service = services[pattern]
-        return service
+        try:
+            service = services[pattern]
+        except KeyError:
+            return None
+        else:
+            return service
```

Comment:
Did you check that there is no usage of current_service() still catching keyerror?